drop-storage/src/types.rs

```diff
@@ -0,0 +1,192 @@
+use serde::Serialize;
+
+type TransferId = String;
+type FilePath = String;
+
+#[derive(Debug, Copy, Clone)]
+pub enum TransferType {
+    Incoming = 0,
+    Outgoing = 1,
+}
+
+impl From<TransferType> for i32 {
+    fn from(transfer_type: TransferType) -> Self {
+        match transfer_type {
+            TransferType::Incoming => 0,
+            TransferType::Outgoing => 1,
+        }
+    }
+}
```

Comment:
This is very explicit and clear. What happens if I sometime later switch the places of members by using repr?

Reply:
It doesn't change because the enum variants already have the explicit discriminants

Reply:
If we have
```
#[repr(u32)]
enum Transfer {
    Outgoing,
    Incoming,
}

```
then it will be converted to a different number given that we switch `Outgoing` and `Incoming`. With the current implementation, we're set in stone to return 0 and 1 explicitly. Also helps with readability a bit.

Reply:
So let's use:
```rust
#[repr(u32)]
enum Transfer {
    Incoming = 0,
    Outgoing = 1,
}
```

Reply:
That's nice as well

Reply:
done